Mathlib/CategoryTheory/FiberedCategory/HomLift.lean

```diff
@@ -113,7 +113,7 @@ lemma of_commSq {R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (ha : p.o
 instance comp {R S T : 𝒮} {a b c : 𝒳} (f : R ⟶ S) (g : S ⟶ T) (φ : a ⟶ b)
     (ψ : b ⟶ c) [p.IsHomLift f φ] [p.IsHomLift g ψ] : p.IsHomLift (f ≫ g) (φ ≫ ψ) := by
   apply of_commSq
-  rw [p.map_comp]
+  on_goal 1 => rw [p.map_comp]
```

Comment:
The line below applies to both goals; this rw just massages the first into the suitable form.

Reply:
I think this is obscure enough that it warrants a comment in the code. I would never have guessed that behaviour if I had to fix this proof because it broke for unrelated reasons.

Reply:
Sure, I've added a comment to this instance and the four other places with similar logic.